op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,358 @@
+		done:       make(chan struct{}),
+	}
+}
+
+// run starts the hub's main loop
+func (h *Hub) run() {
+	for {
+		select {
+		case <-h.done:
+			// Close all client connections
+			h.clientsMu.Lock()
+			for client := range h.clients {
+				client.Close()
+				delete(h.clients, client)
+			}
+			h.clientsMu.Unlock()
+			return
+		case client := <-h.register:
+			h.clientsMu.Lock()
+			h.clients[client] = true
+			h.clientsMu.Unlock()
+		case client := <-h.unregister:
+			h.clientsMu.Lock()
+			if _, ok := h.clients[client]; ok {
+				delete(h.clients, client)
+				client.Close()
+			}
+			h.clientsMu.Unlock()
+		case message := <-h.broadcast:
+			h.clientsMu.RLock()
+			for client := range h.clients {
+				select {
+				case client.send <- message:
+					// Message sent successfully
+					client.failureCount = 0 // Reset failure count on success
+				default:
+					// Client send buffer is full, increment failure count
+					client.failureCount++
+					if client.failureCount >= client.maxFailures {
+						// Send to unregister channel - now non-blocking with buffered channel
+						h.unregister <- client
+					}
+				}
+			}
+			h.clientsMu.RUnlock()
+		}
+	}
+}
+
+// Handler implements the FlashblockHandler interface
+type Handler struct {
+	cfg             Config
+	log             log.Logger
+	isLeaderFn      func(context.Context) bool
+	rollupBoostConn *websocket.Conn
+	server          *http.Server
+	hub             *Hub
+}
+
+// NewHandler creates a new flashblocks handler
+func NewHandler(cfg Config, log log.Logger, isLeaderFn func(context.Context) bool) (FlashblockHandler, error) {
+	// Create the hub
+	hub := newHub()
+
+	// Initialize the handler
+	handler := &Handler{
+		cfg:        cfg,
+		log:        log,
+		isLeaderFn: isLeaderFn,
+		hub:        hub,
+	}
+
+	// Establish connection to rollup boost if URL is configured
+	if cfg.RollupBoostWsURL != "" {
+		log.Info("connecting to rollup boost WebSocket", "url", cfg.RollupBoostWsURL)
+		conn, _, err := websocket.DefaultDialer.Dial(cfg.RollupBoostWsURL, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to connect to rollup boost WebSocket: %w", err)
+		}
+		handler.rollupBoostConn = conn
+		log.Info("connected to rollup boost WebSocket", "url", cfg.RollupBoostWsURL)
+	} else {
+		log.Info("rollup boost WebSocket disabled, no URL configured")
+	}
+
+	return handler, nil
+}
+
+// Start initializes the flashblocks handler and starts the WebSocket server and rollup boost listener
+func (h *Handler) Start(ctx context.Context) error {
+	go h.hub.run()
+
+	// Start the WebSocket server
+	if err := h.startWebSocketServer(ctx); err != nil {
+		return err
+	}
+
+	// Start the rollup boost listener if connection exists
+	if h.rollupBoostConn != nil {
+		go h.listenToRollupBoost(ctx)
+	}
+
+	return nil
+}
+
+// Stop closes any open WebSocket connections and shuts down the server
+func (h *Handler) Stop() {
+	// Signal the hub to stop
+	close(h.hub.done)
+
```

Comment:
```suggestion
	h.hub.done <- struct{}
```

Due to the existing close(), we get the following issues:
- The idempotency of this method gets sacrificed (it might be potentially called twice)
- Can't straightforwardly (re)Start() the stopped handler. Because we'd need to handle the case of h.hub.done channel being potentially closed or not, whenever running `go h.hub.run()`

Or if we want to enforce the assumption that a Handler can't be restarted after a Stop, then I'd prefer having some validations/guards for that under `Start()` against a potentially closed h.hub.Done or a closed Handler altogether  (like you already have some such as `if h.rollupBoostConn != nil {` )

Reply:
I'm not sure if this is necessary, as the existing pattern here does assume that Handler or any other service's `Stop` will only be called during shutdown.